pkg/net/stats.go

```diff
@@ -0,0 +1,74 @@
+package net
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+const windowSize = 16
+
+var (
+	GlobalNetStats   = &Stats{}
+	OneMinWindow     = &Stats{}
+	FiveMinWindow    = &Stats{}
+	FifteenMinWindow = &Stats{}
+
+	slidingWindow = [windowSize]*Stats{}
+	circularIndex = 0
+)
+
+func init() {
+	slidingWindow = [windowSize]*Stats{}
+	for i := range slidingWindow {
+		slidingWindow[i] = &Stats{}
+	}
+	slidingCounter := func() {
```

Comment:
can you write simply `go func()` instead of having a function for `slidingCounter()` does it add any value.. ?